src/OpenTelemetry.Instrumentation.AWS/Implementation/Tracing/AWSTraceSpan.cs

```diff
@@ -59,8 +59,10 @@ public override void RecordException(Exception exception, Attributes? attributes
 
         var tags = attributes != null ? new TagList(attributes.AllAttributes.ToArray()) : default;
 
-        this.activity.RecordException(exception, tags);
+        this.activity.AddException(exception, tags);
+#pragma warning disable CS0618 // Type or member is obsolete
```

Comment:
Where something was calling `ActivityExtensions.SetStatus` I suppressed warning instead of updating to use `Activity.Status` \ `Activity.StatusDescription`. The reason for that is they don't do the same things. In order for `Activity.Status` \ `Activity.StatusDescription` to work, users need to also upgrade to a compatible exporter which understands the status fields on `Activity`. I decided calling the old extension for now was the safer bet.

Reply:
Exception made in `ElasticsearchRequestPipelineDiagnosticListener` which was calling the newer Activity APIs to translate http status code but calling the obsolete extensions when there was an exception. Updated to use the newer Activity APIs there so it is consistent.

Reply:
#2321 created